### PR TITLE
utils/clock: add clock package

### PR DESCRIPTION
**Pull request by axw**
Add a clock package which contains a Clock
interface, and WallClock var whose type
implements Clock in terms of the wall clock.

(Review request: http://reviews.vapour.ws/r/2327/)

### CODE REVIEW

File: clock/wall.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+type wallClock struct{}
+
+// Now is part of the Clock interface.
+func (wallClock) Now() time.Time {
+	return time.Now()
+}
+
+// Alarm returns a channel that will send a value at some point after
+// the supplied time.
+func (clock wallClock) After(d time.Duration) <-chan time.Time {
```
Comment (wallyworld) [inline clock/wall.go]: don't need a named receiver

**Comment by wallyworld** [comment]
LGTM

File: clock/wall.go (hunk 1 above)
Comment (axw) [inline clock/wall.go]: fixed, ta

**Comment by axw** [comment]
$$merge$$

**Pull request merged**